mod_reforged/hooks/skills/perks/perk_anticipation.nut

```diff
@@ -1,119 +1,101 @@
+		this.m.TempDamageReceivedTotalMult = this.getDamageReceivedTotalMult();
+		_properties.DamageReceivedTotalMult *= this.m.TempDamageReceivedTotalMult;
 	}
 
 	q.onDamageReceived <- function( _attacker, _damageHitpoints, _damageArmor )
 	{
-		if (this.m.IsAboutToConsumeUse == false) return;	// We only consume one use for each registered attack. But a single attack that deals damage multiple times will therefor have the damage of all instances reduced
-		this.m.IsAboutToConsumeUse = false;
-
-		this.m.UsesRemaining = ::Math.max(0, this.m.UsesRemaining - 1);
+		if (this.m.TempDamageReceivedTotalMult == 1.0)
+			return;
 
-		if (_attacker == null)	// This can for example happen when this character receives a mortar attack.
-		{
-			::Tactical.EventLog.logEx(::Const.UI.getColorizedEntityName(this.getContainer().getActor()) + " anticipated an attack, reducing damage received by " + ::MSU.Text.colorPositive(this.m.TempDamageReduction + "%"));
-		}
-		else
-		{
-			::Tactical.EventLog.logEx(::Const.UI.getColorizedEntityName(this.getContainer().getActor()) + " anticipated the attack of " + ::Const.UI.getColorizedEntityName(_attacker) + ", reducing damage received by " + ::MSU.Text.colorPositive(this.m.TempDamageReduction + "%"));
-		}
+		// _attacker can be null e.g. when receiving damage from mortar
+		::Tactical.EventLog.logEx(format("%s anticipated %s, reducing damage received by %s", ::Const.UI.getColorizedEntityName(this.getContainer().getActor()), _attacker == null ? "an attack" : "the attack of " + ::Const.UI.getColorizedEntityName(_attacker), ::MSU.Text.colorPositive(this.m.TempDamageReduction + "%")));
 	}
 
-
-	// private functions
-	q.isEnabled <- function()
+	q.getDamageReceivedTotalMult <- function( _attacker = null )
 	{
-		return (this.m.UsesRemaining > 0);
+		return ::Math.maxf(this.m.MinDamageReceivedTotalMult, 1.0 - this.getContainer().getActor().getCurrentProperties().getRangedDefense() * 0.01);
 	}
 
-	// returns a value between 0 and 100 indicitation the damage reduction in %
-	q.calculateDamageReduction <- function( _attacker = null )
+	function getChance( _attacker = null )
 	{
-		local damageReduction = this.m.DamageReductionBase;
-		damageReduction += ::Math.max(0, this.getContainer().getActor().getCurrentProperties().getRangedDefense());
+		local actor = this.getContainer().getActor();
+		local ret = this.m.BaseChance + actor.getRangedDefense();
 
-		if (_attacker != null && _attacker.isPlacedOnMap() && this.getContainer().getActor().isPlacedOnMap())
+		if (_attacker != null && _attacker.isPlacedOnMap() && actor.isPlacedOnMap())
 		{
-			// -1 because adjacent entities count as a distance of 1 but we want only the number of tiles between the entities
-			damageReduction += this.m.DamageReductionPerTile * (_attacker.getTile().getDistanceTo(this.getContainer().getActor().getTile()) - 1);
+			ret += this.m.ChancePerTile;
```

Comment:
You always add 10 to the return value.
Instead you want to calculate the tile distance, e.g. like I had it before:
`_attacker.getTile().getDistanceTo(this.getContainer().getActor().getTile()) - 1`